vcenter_operator/phelm.py

```diff
@@ -48,6 +52,21 @@ class DeploymentState(object):
     items = attr.ib(default=attr.Factory(OrderedDict))
     actions = attr.ib(default=attr.Factory(OrderedDict))
 
+    def render(self, scope, options):
+        try:
+            template_names = env.list_templates(
+                filter_func=lambda x: (x.startswith(scope)
+                                       and x.endswith('.yaml.j2')))
+            for template_name in template_names:
+                try:
+                    template = env.get_template(template_name)
+                    result = template.render(options)
+                    self.add(result)
+                except (TemplateError, YAMLError):
+                    LOG.exception("Failed to render %s", template_name)
+        except TemplateLoadingFailed:
+            LOG.exception("Failed to load templates")
```

Comment:
With catching this exception moving into the `DeploymentState`, we would still have whatever state is in the `DeploymentState` already after this function, while it would previously break out of the current apply-loop for all vcenters.
So if the template loading failed only for `vcenter_datacenter`, we would still contain the `vcenter_cluster` state added before. This could lead to the vcenter-operator deleting a lot of deployments. I think we need to fix this and bubble up some Exception that we can handle gracefully in `poll()`.

Reply:
Good point, I'll let the exception bubble up again